src/ast/table.rs

```diff
@@ -126,6 +127,86 @@ impl<'a> Table<'a> {
 
         Ok(result)
     }
+
+    pub fn left_join<J>(mut self, join: J) -> Self
+    where
+        J: Into<JoinData<'a>>,
+    {
+        match self.typ {
+            TableType::Table(table_name) => {
+                self.typ = TableType::JoinedTable((table_name, vec![Join::Left(join.into())]))
+            }
+            TableType::JoinedTable((_, ref mut joins)) => joins.push(Join::Left(join.into())),
+            TableType::Query(_) => {
+                panic!("You cannot left_join on a table of type Query")
+            }
+            TableType::Values(_) => {
+                panic!("You cannot left_join on a table of type Values")
+            }
+        }
+
+        self
+    }
+
+    pub fn inner_join<J>(mut self, join: J) -> Self
```

Comment:
This needs a public documentation with examples. Read Select for info how they should be written.